src/broker/cas_execute.c

```diff
@@ -1075,7 +1075,38 @@ ux_cgw_prepare (char *sql_stmt, int flag, char auto_commit_mode, T_NET_BUF * net
   srv_handle->schema_type = -1;
   srv_handle->auto_commit_mode = auto_commit_mode;
 
-  ALLOC_COPY (srv_handle->sql_stmt, sql_stmt);
+  if (cgw_get_dbms_type () == SUPPORTED_DBMS_ORACLE)
+    {
+      char *is_cublink = NULL;
+      is_cublink = strstr (sql_stmt, REWRITE_DELIMITER_CUBLINK);
+      if (is_cublink != NULL)
+	{
+	  char *rewrite_sql = NULL;
+	  err_code = cgw_rewrite_query (sql_stmt, &rewrite_sql);
+	  if (err_code == -1)
+	    {
+	      err_code = ERROR_INFO_SET (db_error_code (), DBMS_ERROR_INDICATOR);
+	      goto prepare_error;
+	    }
+	  else if (err_code == -2)
```

Comment:
Direct use of negative numbers reduces readability.
Is it difficult to use a symbolic constant like REWRITE_FAILED?